src/tests/stats/test_distribution_normal.fypp

```diff
@@ -0,0 +1,235 @@
+                             (1.08284164094813181722365413861552952_${k1}$,    &
+                             0.277168639672963013076412153168348595_${k1}$),   &
+                             (1.41924946329521489696290359461272601_${k1}$,    &
+                             0.498445561155580918466512230224907398_${k1}$),   &
+                             (1.72639126368764062036120776610914618_${k1}$,    &
+                             0.715802936564464420410303091557580046_${k1}$),   &
+                             (1.98950590834134349860207180427096318_${k1}$,    &
+                             0.115721315405046931701349421928171068_${k1}$),   &
+                            (-1.16929014824793620075382705181255005_${k1}$,    &
+                             0.250744737486995217246033007540972903_${k1}$),   &
+                             (1.57160542831869509683428987045772374_${k1}$,    &
+                             0.638282596371312238581197107123443857_${k1}$),   &
+                            (-1.36106107654239116833139178197598085_${k1}$,    &
+                             0.166259201494369124318950525776017457_${k1}$),   &
+                             (1.13403096805387920698038328737311531_${k1}$,    &
+                              1.04232618148691447146347854868508875_${k1}$),   &
+                            (-1.68220535920475811053620418533682823_${k1}$,    &
+                              1.63361446685040256898702182297711261_${k1}$)]
+        #:endif
+
+        print *, "Test normal_distribution_rvs_${t1[0]}$${k1}$"
+        seed = 25836914
+        call random_seed(seed, get)
+        #:if t1[0] == "r"
+            #! for real type
+        loc = 0.5_${k1}$; scale = 2.0_${k1}$
+        #:else
+            #! for complex type
+        loc = (0.5_${k1}$, 1.0_${k1}$); scale = (1.5_${k1}$, 0.5_${k1}$)
+        #:endif
+
+        do i = 1, k
+           res(i) = nor_rvs(loc, scale)     ! 2 dummies
+        end do
+        res(6:10) = nor_rvs(loc, scale, k)  ! 3 dummies
+        call check(all(abs(res - ans) < ${k1}$tol),                            &
+            msg="normal_distribution_rvs_${t1[0]}$${k1}$ failed", warn=warn)
+    end subroutine test_nor_rvs_${t1[0]}$${k1}$
+
+    #:endfor
+
+
+
+
+    #:for k1, t1 in RC_KINDS_TYPES
+    subroutine test_nor_pdf_${t1[0]}$${k1}$
+
+        ${t1}$ :: x1, x2(3,4), loc, scale
+        integer, parameter :: k = 5
+        integer :: i, n, seed, get
+        real :: res(3,5)
+        #:if t1[0] == "r"
+            #! for real type
+        real, parameter :: ans(15) =                                           &
+                         [0.215050772, 0.215050772, 0.215050772, 0.200537622,  &
+                          5.66161536E-02, 0.238986954, 0.265935957,0.262147546,&
+                          0.249866411, 3.98721099E-02, 0.265902370,0.161311597,&
+                          0.249177739, 0.237427220, 0.155696079]
```

Comment:
I think here, if the numbers are updated with quad-precision results, then the tests should be able to catch errors caused by loss of precision. They have a precision-dependent tolerance already - it might need tweaking if extended to higher precisions, but looks like it should just work.